src/Workspaces/CSharp/Portable/Rename/CSharpRenameRewriterLanguageService.cs

```diff
@@ -230,6 +231,19 @@ public override SyntaxToken VisitToken(SyntaxToken token)
                 return newToken;
             }
 
+            /// <summary>
+            /// Returns whether the given <see cref="SyntaxToken"/> could possibly be a destructor
+            /// conflict with the given replacementText. This is a heuristic that could introduce
+            /// conflict check tokens unnecessarily where a bitwise complement is applied to an
+            /// identifier.
+            /// </summary>
+            private bool IsPossiblyDestructorConflict(SyntaxToken token, string replacementText)
+            {
+                return _replacementText == "Finalize" &&
+                    token.IsKind(SyntaxKind.IdentifierToken) &&
+                    token.GetPreviousToken().IsKind(SyntaxKind.TildeToken);
```

Comment:
I'm confused.  Are you trying to say "is this token the identifier token of a destructor"?  If so, why not just check the parent of the token and check it's kind?